app-android/src/main/java/com/ustadmobile/port/android/view/CourseDiscussionDetailFragment.kt

```diff
@@ -145,6 +185,124 @@ class CourseDiscussionDetailFragment: UstadDetailFragment<CourseDiscussion>(),
             ustadFragmentTitle = value?.courseDiscussionTitle
         }
 
+}
 
 
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun CourseDiscussionDetailScreen(
+    uiState: CourseDiscussionDetailUiState = CourseDiscussionDetailUiState(),
+    onClickAddPost: () -> Unit = {},
+    onClickPost: (DiscussionPostWithDetails) -> Unit = {},
+    onClickDeletePost: (DiscussionPostWithDetails) -> Unit = {},
+){
+
+    LazyColumn{
+
+        //Description:
+        item{
+
+            Text(
+                uiState.courseDiscussion?.courseDiscussionDesc?:"",
+                style = Typography.body1,
+                modifier = Modifier.padding(8.dp)
+            )
+        }
+
+        item{
+            Spacer(modifier = Modifier.height(10.dp))
+        }
+
+        item{
+            Text(stringResource(R.string.posts),
+                style = Typography.h4,
+                modifier = Modifier.padding(8.dp))
+        }
+
+
+        items(
+            items = uiState.posts,
+            key = { post -> post.discussionPostUid }
+        ){ post ->
+
+            val context = LocalContext.current
+            val datePosted = remember { DateFormat.getDateFormat(context)
+                .format(Date(post.discussionPostStartDate ?: 0)).toString() }
+
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickPost(post)
+                },
+                icon = {
+                    //TODO: replace with avatar
+                    Icon(
+                        imageVector = Icons.Filled.Person,
+                        contentDescription = null
+                    )
+                },
+                text = {
+                    Text(post.discussionPostMessage ?: "" )
+                },
+                secondaryText = {
+                    Column {
+                        Text(post.discussionPostTitle?: "")
```

Comment:
Shouldn't the title be in the primary text, then the latest message is in the secondary text? I don't think we need the post message itself.